atc/api/resourceserver/check.go

```diff
@@ -55,6 +55,7 @@ func (s *Server) CheckResource(dbPipeline db.Pipeline) http.Handler {
 			dbResourceTypes,
 			reqBody.From,
 			true,
+			!reqBody.Shallow,
```

Comment:
"Shallow" means check the resource itself. In that case, we suppose to not check resource types, then skipIntervalRecursively should to true. Not sure if my understanding is correct or not.